kittysan.js

```diff
@@ -0,0 +1,170 @@
+      previousPage.addEventListener('click', () => {
+        bunny.changeQuery(bunny.query, bunny.category, bunny.page - 1);
+        fetchData();
+      })
+      // add link to search pagination
+      this.searchPagination.appendChild(previousPage);
+    }
+
+    // add all pages
+    for (let i = 1; i <= data.total_pages; i++) {
+      // create link element
+      const link = document.createElement('a');
+      // set active page class
+      if (data.page === i) link.classList.add('active');
+      // add text content
+      link.textContent = i;
+      // handle click event to fetch data from page
+      link.addEventListener('click', () => {
+        bunny.changeQuery(bunny.query, bunny.category, i);
+        fetchData();
+      });
+      // add link to search pagination
+      this.searchPagination.appendChild(link);
+    }
+
+    // add next page
+    if (data.page < data.total_pages) {
+      // create link element
+      const nextPage = document.createElement('a');
+      // add inner html content
+      nextPage.innerHTML = '&raquo;';
+      // handle click event to fetch results from next page
+      nextPage.addEventListener('click', () => {
+        bunny.changeQuery(bunny.query, bunny.category, bunny.page + 1);
+        fetchData();
+      })
+      // add link to search pagination
+      this.searchPagination.appendChild(nextPage);
+    }
+  }
+
+  paintDialog(result) {
+    // log data
+    console.log(result);
+    // add backdrop
+    this.mediaBackdrop.setAttribute('src', result.backdrop_path ? `https://image.tmdb.org/t/p/w500${result.backdrop_path}` : 'http://via.placeholder.com/500x300');
+    // clear existing media ratings
+    while (this.mediaRating.firstChild) {
+      this.mediaRating.removeChild(this.mediaRating.firstChild);
+    }
+    // helper functions
+    const roundHalf = (num) => Math.round(num * 2) * 0.5;
+    const addStar = (cls) => {
+      const icon = document.createElement('i');
+      icon.className = cls;
+      this.mediaRating.appendChild(icon);
+    }
+    // add ze stars
+    const stars = roundHalf(result.vote_average * 0.5);
+    for (let i = 1; i <= 5; i++) {
+      if (i <= stars) {
+        addStar('fas fa-star');
+      } else if (i - 0.5 === stars) {
+        addStar('fas fa-star-half');
+      }
+    }
+    // clear existing media genres
+    while (this.mediaGenres.firstChild) {
+      this.mediaGenres.removeChild(this.mediaGenres.firstChild);
+    }
+    // add genres
+    result.genre_ids.forEach(genre => {
+      if ((bunny.category === 'movie' && bunny.movie_genres[genre]) || (bunny.category === 'tv' && bunny.tv_genres[genre])) {
+        // create genre element
+        const span = document.createElement('span');
+        // add classes
+        span.classList.add('genre');
+        // add text content
+        span.textContent = bunny.category === 'movie' ? bunny.movie_genres[genre].name : bunny.tv_genres[genre].name;
```

Comment:
Take a closer look at this line. Think about where the genre IDs are coming from and what doing a lookup on an array like that is doing. Remember movie_genres is an array of objects. What is the unique identifier here?